packages/vscode-extension/src/plugins/network/network-plugin.ts

```diff
@@ -1,26 +1,57 @@
 import http, { Server } from "http";
+import { spawn } from "child_process";
+import path from "path";
 import { commands, Disposable, window } from "vscode";
 import { WebSocketServer, WebSocket } from "ws";
 import { Devtools } from "../../project/devtools";
 import { ToolKey, ToolPlugin } from "../../project/tools";
 import { extensionContext } from "../../utilities/extensionContext";
+
+import { Logger } from "../../Logger";
 import { NetworkDevtoolsWebviewProvider } from "./NetworkDevtoolsWebviewProvider";
 
 export const NETWORK_PLUGIN_ID = "network";
 
+function startViteServer(onReady: () => void) {
```

Comment:
NIT: I think it's more natural to return a `Promise` from this function, which resolves once the server is ready, and `await` it wherever we call this.